text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+conventions</a> for attributes.
+    </td>
+   </tr>
+</table>
+
+### Minimally Sufficient Id
+
+Often a number of attributes of an entity is readily available for the telemetry
+producer to compose an Id from. Of the available attributes the Entity Id should
+include the minimal set of attributes that is sufficient for uniquely identifying
+that entity. No superfluous attributes should be included in the Id set. For example
+a Process on a host can be uniquely identified by `process.id` attribute. Adding for
+example `process.executable.name` attribute to the Id is unnecessary and violates the
+Minimally Sufficient Id rule.
+
+### Examples of Entities
+
+_This section is non-normative and is present only for the purposes of demonstrating
+the data model._
+
+Here are examples of entities, the typical identifying attributes they
+have and some examples of non-identifying attributes that may be
+associated with the Entity.
+
+<table>
+   <tr>
+    <td><strong>Entity</strong>
+    </td>
+    <td><strong>Entity Type</strong>
+    </td>
+    <td><strong>Identifying Attributes</strong>
+    </td>
+    <td><strong>Non-identifying Attributes</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Service
+    </td>
+    <td>"service"
+    </td>
+    <td>service.name (required)
+<p>
+service.instance.id
+<p>
+service.namespace
+    </td>
+    <td>service.version
+    </td>
+   </tr>
+   <tr>
+    <td>Host
+    </td>
+    <td>"host"
+    </td>
+    <td>host.id
+    </td>
+    <td>host.name
+<p>
+host.type
+<p>
+host.image.id
+<p>
+host.image.name
+    </td>
+   </tr>
+   <tr>
+    <td>K8s Pod
+    </td>
+    <td>"k8s.pod"
+    </td>
+    <td>k8s.pod.uid (required)
+<p>
+k8s.cluster.name
+    </td>
+    <td>Any pod labels
+    </td>
+   </tr>
+   <tr>
+    <td>K8s Pod Container
+    </td>
+    <td>"container"
+    </td>
+    <td>k8s.pod.uid (required)
+<p>
+k8s.cluster.name
+<p>
+container.name
+    </td>
+    <td>Any container labels
+    </td>
+   </tr>
+</table>
+
+See more examples showing nuances of ID field composition in the
+[Entity Identification](#entity-identification) section.
+
+## Entity Events
+
+Information about Entities can be produced and communicated using 2
+types of Entity events: EntityState and EntityDelete.
+
+### EntityState Event
+
+The EntityState event stores information about the _state_ of the Entity
+at a particular moment of time. The data model of the EntityState event
+is the same as the Entity Data model with some extra fields:
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time since when the entity state is described by this event.
+The time is measured by the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Interval
+    </td>
+    <td>milliseconds
```

Comment:
Are we sure sub-millisecond intervals won't ever be useful? On the consumer side it might be easier anyways to just have nanoseconds here so that Timestamp+Interval does not require a conversion.

Reply:
Added a TODO to discuss separately: https://github.com/orgs/open-telemetry/projects/85/views/1?pane=issue&itemId=63753636